app/controllers/spree/admin/promotion_rules_controller.rb

```diff
@@ -49,6 +49,6 @@ def validate_promotion_rule_type
   end
 
   def promotion_rule_params
-    params[:promotion_rule].permit!
+    params[:promotion_rule].permit(:type)
```

Comment:
@aleksandarpetrushev what about `user_id` or `product_group_id` attributes for this model? or preferences?

Reply:
@damianlegawiec we have a similar situation such as in payment_methods_controller. promotion_rules_controller is used only for #create and #destroy. When creating a promotion rule, we know the type only. The additional attributes are sent in PromotionsController#update as nested "promotion_rules_attributes", after the promotion rules have been created. Tested for all promotion rules, only the type is sent.